packages/core/src/shared/tools.ts

```diff
@@ -447,6 +447,10 @@ export class Tools {
         return isLegalUrl(url);
     }
 
+    static normalizeUrl(url: string) {
```

Comment:
Do not add more methods to the `Tools` class. It is considered a technical dept to wrap util functions in a class.